### DISCUSSION TITLE
ESP32, LEDC ledcWriteTone fuction

### DISCUSSION THREAD

**Discussion by guoquan1963** [espressif/arduino-esp32#7110] [General]
Hi,

I am using "ledcWriteTone" to change the frequency of PWM and found that the frequency is not stable...
I checked on website and saw "ESP32, frequent calls to ledcWriteTone function will lead to inaccurate output".

It is fixed or there is ant suggestion?
Can I use "ledcChangeFrequency(PWM1_Ch, (PWM1_Freq/PWM1_DutyCycle), 8);" to do so? I tried it and get error "'ledcChangeFrequency' was not declared in this scope". 

How to fix this error?
Thanks,
Christie

**Comment by guoquan1963** [comment]
Hi,

I want to generate 10Hz(100ms)~~100Hz(10ms) PWM signal. Need to have 0.01ms(accuracy) resolution.

I can update the frequency in real-time without any interference…

When I try to use "ledcWriteTone", it causes the interference based on how fast I update it...

Please let me know how to configure LEDC PWM output?

Thanks,

Christie

**Comment by VojtechBartoska** [comment]
Please, Can you help with answer @P-R-O-C-H-Y?

**Comment by guoquan1963** [comment]
Hi,

I tried ""ledcWrite" to change the duty of PWM on ESP32-WROOM-32E and found the resolution is 10 only even when I set ti to 16...
Can anybody tell me how to change the resolution?
Thanks,
Christie

**Comment by guoquan1963** [comment]
Hi,
Is there anyone to help me to fix the problem? Looks like the following link show the problem.
" https://esp32.com/viewtopic.php?t=18350"
Most likely, it is related to "ledc_set_frequency" function...
I need to change frequency glitchless...Please help me to fix it in Arduino

Thank you very much...

**Reply by P-R-O-C-H-Y** [reply]
@guoquan1963 Hi, are you still having the issues?

**Comment by chibiconsulting** [comment]
Does the ledcWriteTone have an associated method to turn off the tone generation?
It sounds like using ledcWriteTone(TONE_PWM_CHANNEL, 0); breaks further calls to ledcWriteTone(TONE_PWM_CHANNEL, NEW_FREQUENCY);
Perhaps it needs to turn off the tone completely and start over when issuing ledcWriteTone(TONE_PWM_CHANNEL, NEW_FREQUENCY);
